libs/datamodel/core/src/transform/ast_to_dml/validation_pipeline/validations/relations/one_to_one.rs

```diff
@@ -243,3 +243,26 @@ pub(crate) fn back_relation_arity_is_optional(relation: InlineRelationWalker<'_,
         ));
     }
 }
+
+pub(crate) fn fields_and_references_on_wrong_side(relation: InlineRelationWalker<'_, '_>, ctx: &mut Context<'_>) {
+    let (forward, back) = match (relation.forward_relation_field(), relation.back_relation_field()) {
+        (Some(forward), Some(back)) if ctx.diagnostics.errors().is_empty() => (forward, back),
+        _ => return,
+    };
+
+    if forward.is_required() && (back.referencing_fields().is_some() || back.referenced_fields().is_some()) {
+        let message = format!(
+            "The relation field `{back_model}.{back_field}` defines the `fields` and/or `references` argument. Since the other side of the relation — `{forward_model}.{forward_field}` — is required, the `fields` and `references` arguments have to be defined there, otherwise the constraints could not be enforced.",
```

Comment:
It'd be great if we could remove the indirection that "`the fields and references arguments have to be defined`**`there`**" brings. How about something like that? 👇

> The relation field `{back_model}.{back_field}` defines the `fields` and/or `references` argument. You must set them on the opposite required side `{forward_model}.{forward_field}` so that constraints can be enforced.

Reply:
Makes sense! Will do that

Reply:
Done